lib/nameable.rb

```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, "Subclasses has not implemented method 'correct_name'"
+  end
```

Comment:
[OPTIONAL] There's a way you can make the error message more specific:

```
... `"#{self.class} has not implemented method '#{__method__}'"`
```

- `#{self.class} ` will automatically grab the class name from which the error is coming.
-  `#{__method__}` will specify in method the error is occurring thus giving more context of error

Reply:
Hi @KDlamini I have made the error message more specific thanks